### PR TITLE
Fix hashing of datetime-like numpy arrays

### LINKED ISSUE #188: Cannot hash datetime64
```
In [59]: joblib.hashing.hash(numpy.zeros(dtype="<m8[s]", shape=(10,)))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-59-8b9b8a0f034d> in <module>()
----> 1 joblib.hashing.hash(numpy.zeros(dtype="<m8[s]", shape=(10,)))

/export/data/home/gholl/venv/bleeding/lib/python3.4/site-packages/joblib/hashing.py in hash(obj, hash_name, coerce_mmap)
    199     else:
    200         hasher = Hasher(hash_name=hash_name)
--> 201     return hasher.hash(obj)

/export/data/home/gholl/venv/bleeding/lib/python3.4/site-packages/joblib/hashing.py in hash(self, obj, return_digest)
     51     def hash(self, obj, return_digest=True):
     52         try:
---> 53             self.dump(obj)
     54         except pickle.PicklingError as e:
     55             warnings.warn('PicklingError while hashing %r: %r' % (obj, e))

/usr/lib64/python3.4/pickle.py in dump(self, obj)
    408         if self.proto >= 4:
    409             self.framer.start_framing()
--> 410         self.save(obj)
    411         self.write(STOP)
    412         self.framer.end_framing()

/export/data/home/gholl/venv/bleeding/lib/python3.4/site-packages/joblib/hashing.py in save(self, obj)
    162                 # Python 3.2's memoryview raise a ValueError instead of a
    163                 # TypeError or a BufferError
--> 164                 self._hash.update(self._getbuffer(obj.flatten()))
    165 
    166             # We store the class, to be able to distinguish between

ValueError: cannot include dtype 'm' in a buffer
```

### PR DESCRIPTION
**Pull request by lesteve**
Fixes #188.

### CODE REVIEW

**Comment by lesteve** [comment]
Forgot to say, but I checked that this fix didn't affect the hash of non datetime-like numpy arrays.

File: joblib/hashing.py [hunk 1]
```diff
@@ -154,14 +154,19 @@ def save(self, obj):
         if isinstance(obj, self.np.ndarray) and not obj.dtype.hasobject:
             # Compute a hash of the object:
             try:
-                self._hash.update(self._getbuffer(obj))
+                # memoryview is not supported for some dtypes,
+                # e.g. datetime64. The workaround is to view the array as
+                # bytes before taking the memoryview
```
Comment (ogrisel) [inline joblib/hashing.py]: Maybe you could add a link the relevant numpy issue in the comment.

**Comment by ogrisel** [comment]
Other than my comment, LGTM.

**Comment by GaelVaroquaux** [comment]
Great solution. I like it very much. LGTM. Please merge after addressing @ogrisel 's comment.

**Pull request merged**